pandas/tests/frame/methods/test_rank.py

```diff
@@ -507,7 +490,9 @@ def test_rank_string_dtype(self, string_dtype_no_object):
         # GH#55362
         obj = Series(["foo", "foo", None, "foo"], dtype=string_dtype_no_object)
         result = obj.rank(method="first")
-        exp_dtype = "Int64" if string_dtype_no_object.na_value is pd.NA else "float64"
+        exp_dtype = (
+            "Float64" if string_dtype_no_object == "string[pyarrow]" else "float64"
```

Comment:
Why does this change the `string[python]` case to `float64` from `Float64`? I guess yet another discussion for PDEP-13, but shouldn't the `if string_dtype_no_object.na_value` invariant still remain?

Reply:
To be clear this PR doesn't "change" anything regarding the flavor of dtypes (just changing int to float) and is only testing what we have right now, but you are certainly right this is inconsistent.

The technical reason is (AFAIK) that we just never implemented (yet) `rank` for our nullable dtypes in general (so also the nullable int returns numpy floats instead of nullable float). This can be considered as a missing part of the implementation. 
And at some point we added a custom `rank` implementation on ArrowExtensionArray using pyarrow compute, and because of the subclass structure, the ArrowStringArray (i.e. `"string[pyarrow]"`) inherits that but converts the pyarrow result to a nullable dtype (and `"string[python]"` array class does not inherit from this, so uses the base class `rank` implementation with always returns numpy types).

Reply:
Ah OK...very confusing. Thanks for clarifying